src/broadcast/participant.rs

```diff
@@ -168,14 +168,10 @@ impl BroadcastParticipant {
     ) -> Result<Option<BroadcastOutput>> {
         info!("Processing broadcast vote.");
 
-        let mut message_votes: HashMap<BroadcastIndex, Vec<u8>> =
-            match self
-                .storage
-                .retrieve(StorableType::BroadcastSet, sid, self.id())
-            {
-                Ok(a) => deserialize!(&a)?,
-                Err(_) => HashMap::new(),
-            };
+        let mut message_votes: HashMap<BroadcastIndex, Vec<u8>> = self
```

Comment:
Note! This behavior is _not_ identical to the previous version because this will work if retrieval succeeds but deserialization fails (whereas the previous version would error our if deserialization fails). Is this OK? I think so, because this info is created locally anyways, so should always properly deserialize. However, if this is an issue we should revisit this, maybe with a `retrieve_serialized` method.

Reply:
This behavior occurs in other places in the code as well.

Reply:
I see our serialization/deserialization as temporary. And I agree that a correct implementation of this code should never have deserialization fail. So I think its fine.